The `platform/windows/detect.py` seems to break MSVC detection on CI it seems, it falls back partially to MinGW (but keeping MSVC parameters) and fails:

```
scons: Building targets ...
run(["core\object\gdvirtual.gen.inc"], ["core\object\make_virtuals.py"])
Executing builder function in subprocess: module_path='D:\\a\\godot\\godot\\core\\object\\make_virtuals.py', parameter_file='C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\7f6697c8b1d243b8b26da514830bd521.json', parameter_file_size=17774, target=['D:\\a\\godot\\godot\\core\\object\\gdvirtual.gen.inc'], source=['D:\\a\\godot\\godot\\core\\object\\make_virtuals.py']
build_rd_headers(["servers\rendering\renderer_rd\shaders\blit.glsl.gen.h"], ["servers\rendering\renderer_rd\shaders\blit.glsl"])
Executing builder function in subprocess: module_path='D:\\a\\godot\\godot\\glsl_builders.py', parameter_file='C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\302887ed57094ab2979a18b070002895.json', parameter_file_size=17834, target=['D:\\a\\godot\\godot\\servers\\rendering\\renderer_rd\\shaders\\blit.glsl.gen.h'], source=['D:\\a\\godot\\godot\\servers\\rendering\\renderer_rd\\shaders\\blit.glsl']
x86_64-w64-mingw32-g++ /Foplatform\windows\godot_windows.windows.opt.tools.x86_64.obj /c platform\windows\godot_windows.cpp -std=gnu++17 /TP -Wno-error=cpp /nologo -O2 -O2 -mwindows -Wall -Wshadow-local -Wno-misleading-indentation -Werror /DDEBUG_ENABLED /DNO_EDITOR_SPLASH /DWINDOWS_ENABLED /DWASAPI_ENABLED /DWINMIDI_ENABLED /DWINVER=0x0601 /D_WIN32_WINNT=0x0601 /DVULKAN_ENABLED /DGLES3_ENABLED /DMINGW_ENABLED /DMINGW_HAS_SECURE_API=1 /DTOOLS_ENABLED /DMINIZIP_ENABLED /DZSTD_STATIC_LINKING_ONLY /DUSE_VOLK /DVK_USE_PLATFORM_WIN32_KHR /DGLAD_ENABLED /DGLES_OVER_GL /Ithirdparty\freetype\include /Ithirdparty\libpng /Ithirdparty\glad /Ithirdparty\volk /Ithirdparty\vulkan /Ithirdparty\vulkan\include /Ithirdparty\zstd /Ithirdparty\zlib /Iplatform\windows /I.
x86_64-w64-mingw32-g++ /Foplatform\windows\crash_handler_windows.windows.opt.tools.x86_64.obj /c platform\windows\crash_handler_windows.cpp -std=gnu++17 /TP -Wno-error=cpp /nologo -O2 -O2 -mwindows -Wall -Wshadow-local -Wno-misleading-indentation -Werror /DDEBUG_ENABLED /DNO_EDITOR_SPLASH /DWINDOWS_ENABLED /DWASAPI_ENABLED /DWINMIDI_ENABLED /DWINVER=0x0601 /D_WIN32_WINNT=0x0601 /DVULKAN_ENABLED /DGLES3_ENABLED /DMINGW_ENABLED /DMINGW_HAS_SECURE_API=1 /DTOOLS_ENABLED /DMINIZIP_ENABLED /DZSTD_STATIC_LINKING_ONLY /DUSE_VOLK /DVK_USE_PLATFORM_WIN32_KHR /DGLAD_ENABLED /DGLES_OVER_GL /Ithirdparty\freetype\include /Ithirdparty\libpng /Ithirdparty\glad /Ithirdparty\volk /Ithirdparty\vulkan /Ithirdparty\vulkan\include /Ithirdparty\zstd /Ithirdparty\zlib /Iplatform\windows /I.
=====
platform\windows\godot_windows.cpp:31:10: fatal error: main/main.h: No such file or directory
   31 | #include "main/main.h"
      |          ^~~~~~~~~~~~~
scons: *** [platform\windows\godot_windows.windows.opt.tools.x86_64.obj] Error 1
compilation terminated.

=====
=====
platform\windows\crash_handler_windows.cpp:33:10: fatal error: core/config/project_settings.h: No such file or directory
   33 | #include "core/config/project_settings.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [platform\windows\crash_handler_windows.windows.opt.tools.x86_64.obj] Error 1

=====
scons: building terminated because of errors.
```

--- a/SConstruct
+++ b/SConstruct
@@ -438,19 +438,6 @@ if selected_platform in platform_list:
             )
             env.SetOption("num_jobs", safer_cpu_count)
 
-    if env["compiledb"]:
-        # Generating the compilation DB (`compile_commands.json`) requires SCons 4.0.0 or later.
-        from SCons import __version__ as scons_raw_version
-
-        scons_ver = env._get_major_minor_revision(scons_raw_version)
-
-        if scons_ver < (4, 0, 0):
-            print("The `compiledb=yes` option requires SCons 4.0 or later, but your version is %s." % scons_raw_version)
-            Exit(255)
-
-        env.Tool("compilation_db")
-        env.Alias("compiledb", env.CompilationDatabase())
-
     # 'dev' and 'production' are aliases to set default options if they haven't been set
     # manually by the user.
     if env["dev"]:
@@ -839,6 +826,19 @@ if selected_platform in platform_list:
     if env["vsproj"]:
         env.vs_incs = []
         env.vs_srcs = []
+
+    if env["compiledb"]:
+        # Generating the compilation DB (`compile_commands.json`) requires SCons 4.0.0 or later.
+        from SCons import __version__ as scons_raw_version
+
+        scons_ver = env._get_major_minor_revision(scons_raw_version)
+
+        if scons_ver < (4, 0, 0):
+            print("The `compiledb=yes` option requires SCons 4.0 or later, but your version is %s." % scons_raw_version)
+            Exit(255)
+
+        env.Tool("compilation_db")
+        env.Alias("compiledb", env.CompilationDatabase())
 
     Export("env")
 

